server/src/test/java/org/candlepin/pinsetter/tasks/PopulateHostedDBTaskTest.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (c) 2009 - 2012 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.pinsetter.tasks;
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.*;
+import static org.mockito.Mockito.*;
+
+import org.candlepin.model.Content;
+import org.candlepin.model.ContentCurator;
+import org.candlepin.model.PoolCurator;
+import org.candlepin.model.Product;
+import org.candlepin.model.ProductCurator;
+import org.candlepin.service.ProductServiceAdapter;
+import org.candlepin.test.TestUtil;
+
+import org.junit.Test;
+import org.quartz.JobExecutionContext;
+
+import java.util.HashSet;
+import java.util.LinkedList;
+
+
+
+/**
+ * PopulateHostedDBTaskTest
+ */
+public class PopulateHostedDBTaskTest {
+
+    @Test
+    public void testExecute() throws Exception {
+        // Setup
+        JobExecutionContext jec = mock(JobExecutionContext.class);
+
+        ContentCurator contentCurator = mock(ContentCurator.class);
+        PoolCurator poolCurator = mock(PoolCurator.class);
+        ProductCurator productCurator = mock(ProductCurator.class);
+        ProductServiceAdapter psa = mock(ProductServiceAdapter.class);
```

Comment:
If I had to guess I'd imagine this is virtually impossible for us to actually test locally against a real DB. Is that right?

If this is going to go wrong it'll be the usual slew of hibernate detached entity errors and such, it seems like it might be good to make this an actual DB test (extend DatabaseTestFixture), use real curators for saving, but mock the adapters and the product ID list we're going to import. Then we're closely simulating prod and we get some certainty that the job will actually get the data into the db. Would that be possible?